### PR TITLE
refactor: changed params to Annotated form

### PR DESCRIPTION
**Pull request by rosstitmarsh**
### Pull Request Checklist

- [x] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [x] Pre-Commit Checks were ran and passed
- [x] Tests were ran and passed

### Description

- Updated the Body, Depends and Parameter params to use the Annotated style as shown in the [Litestar documentation](https://docs.litestar.dev/latest/usage/requests.html#validation-and-customization-of-openapi-documentation)
- I had to add `# noqa: TCH002` because Ruff falsely thinks the imports are only needed when type checking

### CODE REVIEW

File: src/app/domain/accounts/controllers/access.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Annotated
 
 from litestar import Controller, MediaType, Response, get, post
 from litestar.di import Provide
-from litestar.enums import RequestEncodingType
-from litestar.params import Body
+from litestar.enums import RequestEncodingType  # noqa: TCH002
```
Comment (cofin) [inline src/app/domain/accounts/controllers/access.py:8]: Do you get an error here without the `noqa`?

**Review by cofin — APPROVED** [review]
Thanks for this.  I left a note on the `TCH002` rule but otherwise looks great.

**Comment by cofin** [comment]
Tests are likely going to fail due to some plugin changes that need to be merged.

File: src/app/domain/accounts/controllers/access.py (hunk 1 above)
Comment (rosstitmarsh) [reply src/app/domain/accounts/controllers/access.py:8]: Yes, Ruff says this without it.
```
src/app/domain/accounts/controllers/access.py:8:28: TCH002 Move third-party import  litestar.enums.RequestEncodingType` into a type-checking block
src/app/domain/accounts/controllers/access.py:9:29: TCH002 Move third-party import `litestar.params.Body` into a type-checking block
```
None of the [settings for flake8-type-checking](https://docs.astral.sh/ruff/settings/#flake8-type-checking) look they are any help.

Comment (rosstitmarsh) [reply src/app/domain/accounts/controllers/access.py]: I have removed the noqa comments with the help of `signature_namespace`

**Pull request merged**